### PR TITLE
feat: prevent crash with multiplet analysis

### PR DESCRIPTION
**Pull request by lpatiny**
close https://github.com/cheminfo/nmrium/issues/1328

### CODE REVIEW

File: src/component/1d/Viewer1D.tsx [hunk 1]
```diff
@@ -150,7 +155,10 @@ function Viewer1D({ emptyText = undefined }: Viewer1DProps) {
           case options.rangesPicking.id:
             dispatch({
               type: ADD_RANGE,
-              ...brushData,
+              payload: {
+                ...brushData,
+                alert,
```
Comment (targos) [inline src/component/1d/Viewer1D.tsx:160]: please fix the missing dependency warning

**Comment by targos** [comment]
@stropitek Please spend some time with @Sebastien-Ahkrin to refactor this so that no side-effects happen in the reducer or in the `data` folder.

File: src/data/data1d/Spectrum1D/ranges/detectSignal.ts [hunk 2]
```diff
@@ -1,26 +1,39 @@
...
+    });
+    if (result && result.chemShift === undefined) return;
+    return {
+      multiplicity: result.js.map((j) => j.multiplicity).join(''),
+      kind: 'signal',
+      delta: result.chemShift,
+      js: result.js,
+    };
+  } else {
+    throw new Error('max lenght is equal to 2048.');
```
Comment (stropitek) [inline src/data/data1d/Spectrum1D/ranges/detectSignal.ts:37]:
```suggestion
    throw new Error('length of signal should not exceed 2048 points');
```

File: src/data/data1d/Spectrum1D/ranges/detectSignal.ts [hunk 3]
```diff
@@ -1,26 +1,39 @@
...
-    symmetrizeEachStep: true,
-  });
-  if (result && result.chemShift === undefined) return;
-  return {
-    multiplicity: result.js.map((j) => j.multiplicity).join(''),
-    kind: 'signal',
-    delta: result.chemShift,
-    js: result.js,
-  };
+  if (toIndex - fromIndex < MAX_LENGTH) {
```
Comment (stropitek) [inline src/data/data1d/Spectrum1D/ranges/detectSignal.ts:18]:
```suggestion
  if (toIndex - fromIndex <= MAX_LENGTH) {
```

**Comment by stropitek** [comment]
Please fix the lintint issues

File: src/component/1d/Viewer1D.tsx [hunk 4]
```diff
@@ -147,12 +156,35 @@ function Viewer1D({ emptyText = undefined }: Viewer1DProps) {
...
+
+            if (!activeSpectrum) break;
+
+            const {
+              data: { x },
+            } = state.data[activeSpectrum.index] as Datum1D;
+
+            const { fromIndex, toIndex } = xGetFromToIndex(x, { from, to });
+
+            if (toIndex - fromIndex < MAX_LENGTH) {
```
Comment (stropitek) [inline src/component/1d/Viewer1D.tsx:173]:
```suggestion
            if (toIndex - fromIndex <= MAX_LENGTH) {
```

**Review by stropitek — APPROVED** [review]

**Pull request merged**